### DISCUSSION TITLE
Sync Spotify Popularity with Navidrome Ratings Using My New Script (sptnr)

### DISCUSSION THREAD

**Discussion by krestaino** [navidrome/navidrome#2773] [Show and tell]
Hey all, I posted this on Reddit, but figured I'd post here too. I wrote this script which syncs Spotify's track popularity to Navidrome star ratings. Why? I don't use the star ratings; I just use the favorite option. So, I was brainstorming ideas to repurpose the star rating and came up with this. I've been using it for the past few months and decided to clean it up and share it. So if this sounds like something you would like, check it out! It's very simple to use, and I've included a public Docker image allowing you to run it with a single command, without any additional configuration or setup. Simply run:
```
docker run -t \
  -e NAV_BASE_URL=your_navidrome_server_url \
  -e NAV_USER=your_navidrome_username \
  -e NAV_PASS=your_navidrome_password \
  -e SPOTIFY_CLIENT_ID=your_spotify_client_id \
  -e SPOTIFY_CLIENT_SECRET=your_spotify_client_secret \
  krestaino/sptnr:latest
 ```
There are a few features which I've documented in the readme, which you can see over at the GitHub repo. If you have any questions, feel free to leave them here. Contributions are welcome, so if you want to help clean up my messy Python code, please do. I don't write Python; this was an exercise to get more comfortable with it. Oh, and I call this script Spotify Popularity to Navidrome Ratings, or "sptnr" for short. I'm not good with coming up with names.

GitHub: https://github.com/krestaino/sptnr
Docker: https://hub.docker.com/r/krestaino/sptnr

|![image](https://github.com/navidrome/navidrome/assets/5562514/6e5e162d-2a70-4391-81d3-d276f8ea03c5)| 
|:--:| 
|Screenshot of the script processing one album and the results in Symfonium|

**Comment by jr0n** [comment]
It would be great if this functionality could be baked into a Navidrome plugin (External Integrations?) and actively maintained. Thanks for making it!
